TiDB Integration: user experience improvements and 2.1.0 release

SabaPing opened the pull request:
### What does this PR do?

This PR includes several improvements for TiDB Integration:

* [Improvement] Update default metrics.
* [Improvement] Update the default dashboard.
* [Improvement] Expose the `metrics` mapper config to let users customize more metrics.
* [Improvement] Let users to customize metric namespace.
* [Improvement] Update readme.
* [Improvement] Add more tests.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [ ] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

SabaPing commented:
Question here: 

I tried to remove the unused dashboard. But the validation CR reported `Error: Short name `overview` at `assets/dashboards` is not allowed to be modified. Please revert to original value.`.

How can I solve this error?

Thanks!

jtappa commented on tidb/DEVELOPMENT.md:
```diff
@@ -2,7 +2,10 @@
+> Make sure you have the latest version of `ddev` installed.
+> Check the version by running `ddev --version`.
+
+2.Install the [Datadog Agent][2].
```
```suggestion
2. Install the [Datadog Agent][2].
```

jtappa commented on tidb/README.md:
```diff
@@ -11,7 +11,7 @@ Connect [TiDB][1] cluster to Datadog in order to:
 >
 > - TiDB 4.0+ is required for this integration. 
-> - Integration of TiDB Cloud with Datadog is not available now.
+> - For TiDB Cloud, please refer to [TiDB Cloud Integration][12].
```
```suggestion
> - For TiDB Cloud, refer to the [TiDB Cloud Integration][12].
```

jtappa commented on tidb/README.md:
```diff
@@ -145,16 +143,15 @@ _Available for Agent versions >6.0_
 
 See [metadata.csv][6] for a list of metrics provided by this check.
 
+> It is possible to use the `metrics` config to collect more other metrics from a TiDB cluster.
```
```suggestion
> It is possible to use the `metrics` config to collect other metrics from a TiDB cluster.
```

coignetp commented on tidb/assets/service_checks.json:
```diff
@@ -1,18 +1 @@
-[
```
Could you add it back? Even though it's not used it's still sent, so it should be documented

coignetp commented on tidb/assets/dashboards/one-screen-overview.json:
```diff
@@ -1 +0,0 @@
-{"title":"TiDB - One-Screen Overview","description":"","widgets":[{"id":7003181654489394,"definition":{"title":"TiDB","type":"group","layout_type":"ordered","widgets":[{"id":1392196187453872,"definition":{"title":"AVG of Get Token Duration","title_size":"16","title_align":"left","show_legend":true,"legend_layout":"auto","legend_columns":["avg","min","max","value","sum"],"time":{},"type":"timeseri ...
```
Is this PR removing this dashboard?

SabaPing replied:
Yes, I tried to remove it but CI checks failed.

SabaPing replied:
I try to keep one dashboard, whose title is `TiDB Overview`, and remove others.

coignetp replied:
We avoid deleting dashboards, do you have any additional context on why this dashboard should be deleted?

SabaPing replied:
Previously, TiDB Integration provided two dashboards, a "cluster overview" dashboard and a supplement to the "cluster overview".

In this new version we have simplified the metrics. with new metrics, the dashboard can be optimized to one. If we can't delete the dashboard, could you give us a better solution? Thanks.

coignetp replied:
It's possible to delete the dashboard, but it's a bit tricky on our side so I prefer to make sure we do that properly for good reasons 😄 
You can remove the dashboard following this https://github.com/DataDog/integrations-extras/pull/1094#discussion_r765553318

coignetp commented on tidb/manifest.json:
```diff
@@ -29,8 +28,8 @@
     "dashboards": {
-      "overview": "assets/dashboards/one-screen-overview.json",
-      "TiDB Overview": "assets/dashboards/overview.json"
+      "TiDB Overview": "assets/dashboards/overview.json",
+      "overview": "assets/dashboards/overview.json"
```
```suggestion
      "TiDB Overview": "assets/dashboards/overview.json"
```
After that you can delete the dashboard file

ChristineTChen commented:
Hi @SabaPing ,

Have you had a chance to try @coignetp's suggestion?

SabaPing commented:
Sorry for long time no update. I'll work on it today and tomorrow.

SabaPing commented:
I followed @coignetp 's suggestion and removed unused dashboard.
I also: 
- updated the remaining dashboard. (This dashboard should be the final version for v2.x.x TiDB Integration)
- improved tests

Thanks a lot!

sarah-witt commented on tidb/metadata.csv:
```diff
@@ -1,43 +1,13 @@
-tidb_cluster.tikv_coprocessor_request_duration_seconds.count,count,,,,Sum of coprocessor request duration,1,tidb,
-tidb_cluster.tidb_session_parse_duration_seconds.sum,count,,,,Total number of session parses,1,tidb,
-tidb_cluster.tidb_session_parse_duration_seconds.count,count,,,,Sum of session parse duration,1,tidb,
+tidb_cluster.tidb_server_handle_query_duration_seconds,count,,,,The distribution of TiDB query durations,1,tidb,
```
```suggestion
tidb_cluster.tidb_server_handle_query_duration_seconds,count,,second,,The distribution of TiDB query durations,1,tidb,
```
It's not required, but could you add units for all of the metrics?

SabaPing replied:
`tidb_cluster.tidb_server_handle_query_duration_seconds` is actually `distribution` type.

How can I define `distribution` metric type? I got an error 

> "distribution" is an invalid metric_type 

when I tried to set type to distribution.

I also checked the validation source code and got this:
```python
VALID_METRIC_TYPE = {'count', 'gauge', 'rate'}
```

sarah-witt replied:
Can you revert back to specifying the two metrics you defined previously? These count and sum metrics are sent by the check for the distribution 
```
tidb_cluster.tidb_server_handle_query_duration_seconds.count,count,,,,The total number of handled queries in server,1,tidb,
tidb_cluster.tidb_server_handle_query_duration_seconds.sum,count,,,,The sum of handled query duration in server,1,tidb,
```

sarah-witt commented on tidb/README.md:
```diff
@@ -145,16 +143,15 @@ _Available for Agent versions >6.0_
 
 See [metadata.csv][6] for a list of metrics provided by this check.
 
+> It is possible to use the `metrics` config to collect other metrics from a TiDB cluster.
```
```suggestion
> It is possible to use the `metrics` configuration option to collect additional metrics from a TiDB cluster.
```

sarah-witt reviewed (commented):
Hi @SabaPing, thanks for the updates! I just left a few more comments

sarah-witt reviewed (commented):
Hi @SabaPing can you also update the PR with the most recent changes in master? This [change](https://github.com/DataDog/integrations-extras/pull/1111) will allow more tests to run on your code

sarah-witt commented on tidb/README.md:
```diff
@@ -181,3 +183,6 @@ Need help? Contact [Datadog support][7].
 [11]: https://github.com/DataDog/integrations-extras/blob/master/tidb/assets/service_checks.json
+[12]: https://docs.datadoghq.com/integrations/tidb_cloud/
+[13]: https://docs.pingcap.com/tidb/stable/tiup-playground
+[14]: 
```
Can you add this link?

jtappa commented on tidb/README.md:
```diff
@@ -158,6 +153,13 @@ See [service_checks.json][11] for a list of service checks provided by this inte
 
+### Missing CPU and Memory metrics for TiKV and TiFlash instances on macOS
+
+You will not get CPU or Memory metrics for TiKV and TiFlash instances in two cases:
+
+- Running TiKV or TiFlash instances with [tiup playground][13] on macOS.
+- Or, running TiKV or TiFlash instances with [docker-compose up][14] on a new Apple M1 machine.
```
```suggestion
CPU and Memory metrics are not provided for TiKV and TiFlash instances in the following cases:

- Running TiKV or TiFlash instances with [tiup playground][13] on macOS.
- Running TiKV or TiFlash instances with [docker-compose up][14] on a new Apple M1 machine.
```

hithwen commented on tidb/manifest.json:
```diff
@@ -29,7 +28,6 @@
     },
     "dashboards": {
-      "overview": "assets/dashboards/one-screen-overview.json",
       "TiDB Overview": "assets/dashboards/overview.json"
```
```suggestion
      "overview": "assets/dashboards/overview.json"
```
This key cannot be changed, but don't worry, the name in the actual overview.json file will be used
